src/store/modules/collection.ts

```diff
@@ -0,0 +1,51 @@
+import { VuexModule, Module, Mutation, Action } from "vuex-module-decorators";
+import {
+  OrderedCollectionPage,
+  ActivityObject,
+  Link
+} from "activitypub-objects";
+
+import client from "apiClient";
+import { AuthenticationUtil } from "@/utils/authentication-util";
+
+const POST_TYPES: string[] = ["Article"];
+
+@Module({ namespaced: true })
+class Collection extends VuexModule {
+  public items: Array<ActivityObject | Link> = [];
```

Comment:
Never mind found the toString method in the Link class :)

Reply:
Either I leave it up to the components how to handle the object or I would have to handle it in the mutation. I would rather the components decide this. What do you think?

Reply:
I would do a "normalization" in the mutation as I would prefer to keep the components as simple as possible and don't spread logic partially in modules and partially in components I guess. Or what's the reason you decided to put the logic into the component?

Reply:
Sorry, I explained that badly. The link is of the type a "link", so it gets its own component, just like the article. Since the link has attributes of a link, I have to put it in a component. At the moment, only articles are allowed on the posts anyway